doc/buffer-watermark/align_watermark_flow_with_port_configuration_test_plan.md

```diff
@@ -0,0 +1,238 @@
+   c. pg-drop: enable the pg-drop counter polling, check whether PG map is generated in COUNTERS_DB, no WATERMARK stats in FLEX_COUNTER_DB
+4. repeat this test 3 times, each time with different type [watermark,queue,pg-drop]
+5. test each type separately and also another test with all of them enabled one after the other with checks in between
+6. repeat the tests with reboot to verify config is saved and same output is generated after reboot
+
+#### Ensure that when only 1 counterpoll enabled only this counterpoll stats exists in DBs
+When only one of the counterpoll types is enabled each time (type[watermark,queue,pg-drop]), make sure only this counterpoll stats exists in:
+- COUNTERS_DB
+- FLEX_COUNTER_DB
+
+### When all relevant counterpoll types enabled stats for all 3 co-exists in DBs
+When all counterpolls types are enabled (type[watermark,queue,pg-drop]), make sure only these counterpoll stats co-exists in:
+- COUNTERS_DB
+- FLEX_COUNTER_DB
+
+### Detailed test flows
+
+#### one relevant counterpoll type enabled with config save and reload
+##### queue enabled one counterpoll config reload
+1. Disable the counter polling for all counters
+2. save and reload configuration
+3. enable the queue counter polling
+4. check whether queue map is generated in COUNTERS_DB
+5. check for QUEUE stats exists in FLEX_COUNTER_DB
+6. no WATERMARK stats in FLEX_COUNTER_DB
+
+##### watermark enabled one counterpoll config reload
+1. Disable the counter polling for all counters
+2. save and reload configuration
+3. enable the watermark counter polling
+4. check whether PG and queue maps are generated in COUNTERS_DB
+5. check for both PG_WATERMARK and QUEUE_WATERMARK stats exists in FLEX_COUNTER_DB
+6. no QUEUE or PG-DROP stats exists
+
+##### pg-drop enabled one counterpoll config reload
+1. Disable the counter polling for all counters
+2. save and reload configuration
+3. enable the pg-drop counter polling
+4. check whether PG map is generated in COUNTERS_DB
+5. check for PG-DROP stats exists in FLEX_COUNTER_DB
+6. no WATERMARK stats in FLEX_COUNTER_DB
+
+
+#### all relevant counterpoll type enabled with config save and reload
+1. Disable the counter polling for all counters
+2. save and reload configuration
+3. enable the queue counter polling
+4. check whether queue map is generated in COUNTERS_DB
+5. check for QUEUE stats exists in FLEX_COUNTER_DB
+6. no WATERMARK stats in FLEX_COUNTER_DB
+7. enable the watermark counter polling
+8. check whether PG and queue maps are generated in COUNTERS_DB
+9. check for both PG_WATERMARK and QUEUE_WATERMARK stats exists in FLEX_COUNTER_DB
+10. no PG-DROP stats exists
+11. enable the pg-drop counter polling
+12. check whether PG map is generated in COUNTERS_DB
+13. check for PG-DROP stats exists in FLEX_COUNTER_DB
+
+
+
+#### one relevant counterpoll type enabled with switch reboot
```

Comment:
we can merge all the test flows in this chapter into `one relevant counterpoll type enabled with config save and reload`. By doing so, we can avoid some duplicate flows